integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,378 @@
+        Deletes all documents with a matching document_ids from the search index.
+
+        :param document_ids: ids of the documents to be deleted.
+        """
+        if self.count_documents == 0:
+            return
+        documents = self._get_raw_documents_by_id(document_ids)
+        if documents:
+            self.client.delete_documents(documents)
+
+    def _get_raw_documents_by_id(self, document_ids: List[str]):
+        """
+        Retrieves all Azure documents with a matching document_ids from the document store.
+
+        :param document_ids: ids of the documents to be retrieved.
+        :returns: list of retrieved Azure documents.
+        """
+        azure_documents = []
+        for doc_id in document_ids:
+            try:
+                document = self.client.get_document(doc_id)
+                azure_documents.append(document)
+            except ResourceNotFoundError:
+                logger.warning(f"Document with ID {doc_id} not found.")
+        return azure_documents
+
+    def get_documents_by_id(self, document_ids: List[str]) -> List[Document]:
+        return self._convert_search_result_to_documents(self._get_raw_documents_by_id(document_ids))
+
+    def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Document]:
+
+        # TODO: Implement this method to filter documents based on metadata fields
+        # For now the implementation is similar to search_documents
+        """
+        Calls the Azure AI Search client's search method and handles pagination.
+        :param search_text: The text to search for. If not supplied, all documents will be retrieved.
+        :returns: A list of Documents that match the given filters.
+        """
+
+        search_text = "*"  # default to search all documents
+        azure_docs = []
+        if filters:
+            # Handle filtering by 'id' first
+            document_ids = filters.get("id")
+            if document_ids:
+                azure_docs = self._get_raw_documents_by_id(document_ids)
+                return self._convert_search_result_to_documents(azure_docs)
+
+            # Handle filtering by 'content'
+            search_text = filters.get("content", "*")
+
+        # Perform search with pagination
+        result = self.client.search(search_text=search_text, top=self.count_documents())
+        azure_docs = list(result)
+        return self._convert_search_result_to_documents(azure_docs)
+
+    def _convert_search_result_to_documents(self, azure_docs: List[Dict[str, Any]]) -> List[Document]:
+
+        documents = []
+        for azure_doc in azure_docs:
+
+            embedding = azure_doc.get("embedding")
+            if embedding == self._dummy_vector:
+                embedding = None
+
+            # Filter out meta fields
+            meta = {
+                key: value
+                for key, value in azure_doc.items()
+                if key not in ["id", "content", "embedding"] and not key.startswith("@")
```

Comment:
@ttmenezes Could you provide some input on this?

The goal is to retrieve all relevant document fields from the index—such as `id`, `content`, `embedding`, and other metadata—while avoiding internal fields like `@search_score`. I haven't found a function that only loads the relevant document fields. While the current implementation seems to work, it might have issues.

Do you have any suggestions for a better solution?

Reply:
Asking our API team if there is a better way to do this, but manually filtering the list as such might be the best option to get this result.